examples/customresources/demos/simplemodule/module.go

```diff
@@ -105,6 +106,19 @@ func (c *counter) DoCommand(ctx context.Context, req map[string]interface{}) (ma
 		// We return the new total after the addition.
 		return map[string]interface{}{"total": atomic.LoadInt64(&c.total)}, nil
 	}
+
+	if cmd == "echo" {
+		// For testing module liveliness
+		return req, nil
+	}
+
+	if cmd == "kill_module" {
+		// For testing module reloading & unexpected exists
+		os.Exit(1)
+		// unreachable return statement needed for compilation
+		return nil, errors.New("unreachable error")
+	}
+
```

Comment:
reviewers, let me know if you don't think simplemodule should include these commands. if so I can make another testmodule for these multiple module scenarios.

Reply:
[testmodule](https://github.com/viamrobotics/rdk/blob/main/module/testmodule/main.go#L108) should already have these commands; could we use that instead? The simple module should probably remain a "simple" example of a golang module.

Reply:
Ah I see you're already using testmodule; was there a reason you also needed simple module to have the same functionality?

Reply:
I got an error related to adding duplicate resource namespaces if I tried to use testmodule twice

Reply:
looks like one more targeted lock gets rid of the race: https://github.com/viamrobotics/rdk/pull/3701/commits/b4d5356cbd3c37125d420ca9ba315ef8c8a05022

Reply:
I'm still pretty against modifying the simple module example to have the echo and kill_module functionality; let me think about the best way to handle starting two modules that have the correct functionality 🤔 .

Reply:
I agree with you, adding more functionality to a "simple" example feels like the opposite of we should be doing :slightly_smiling_face: 

If you can't think of a simple way to improve this, what do you think of me just copying our existing test module and creating a follow-up ticket to enable developers to create test modules as needed?

Reply:
> If you can't think of a simple way to improve this, what do you think of me just copying our existing test module and creating a follow-up ticket to enable developers to create test modules as needed?

That SGTM. I think copying the existing test module and changing the API or model of the helper component should work. I can't think of a super easy to way to write a test utility that "creates" a whole module with the desired functionality, but if that doesn't sound infeasible to you, definitely throw it in a ticket!

Reply:
> I can't think of a super easy to way to write a test utility that "creates" a whole module with the desired functionality, but if that doesn't sound infeasible to you, definitely throw it in a ticket!

Agreed, all the paths I'm considering involve a decent amount of complexity, so best to leave as a follow-up IMO

Reply:
done https://github.com/viamrobotics/rdk/pull/3701/commits/d58d8cb3e1e96b31219b9320b56c611de0e6b23b